### DISCUSSION TITLE
Moving away from MUI, choosing the next UI library

### DISCUSSION THREAD

**Discussion by marcopiovanello** [marcopeocchi/yt-dlp-web-ui#223] [Polls]
Hello!

Last year (almost) I've started an idea of rewriting yt-dlp-webui's frontend with svelte, but the lacking of good UI libraries and the recent transition to svelte 5 made me abandon it.

Some considerations, I love React and I'm planning to stick with it for many more years 😄. I don't understand critics like "React is slow, use library x". Define slow.

But back on the main topic, yt-dlp-webui uses MUI and recently it has become a problem. (Let's be clear, MUI is great I've used in many enterprise projects).

The problem is that the project grew, quite impressively to be honest, MUI cannot be tree-shaken properly (at least in my case), the bundle size is over 500KB just for the layout (event after code-splitting) and many people dislike MUI in terms of UX.

The mobile UX is horrible, mobile was never meant to be the main target of this app, but many users use it from mobile too which it has to be take in cosinderation.

In these 2.5 years many alternatives have emerged.

**Comment by joestump** [comment]
First three all seem like good options. The last two seem costly where cost is your after work hours. 😄

**Reply by marcopiovanello** [reply]
It's indeed true 🥲

**Comment by quadcom** [comment]
I wish I could contribute advice. Alas I'm not a dev and don't know any of the languages you've mentioned. 

I just wanted to leave a note to show that people are seeing this and care. 

Thanks for building this and sharing it with the community!

**Comment by marcopiovanello** [comment]
I think it's almost decided for shadcn/ui!

**Reply by Its4Nik** [reply]
Hey there, just gonna throw my 2 cents into here as well, i recently took a look at https://remix.run it is a react based framework and I think its one of the most user friendly frameworks out there.

Ik it isnt a Component Library but i am using it with ShadCN + Tailwind and the dev experience has been great so far :)

**Reply by marcopiovanello** [reply]
That's great to hear! I don't have experience with Remix, but if it's possible to build client side rendered app it would be great!

For more context: the react frontend needs to be embedded (as static resources) into the binary. Solutions like next, while being great for many reasons, are not feasible.

**Reply by Its4Nik** [reply]
> That's great to hear! I don't have experience with Remix, but if it's possible to build client side rendered app it would be great!
> 
> For more context: the react frontend needs to be embedded (as static resources) into the binary. Solutions like next, while being great for many reasons, are not feasible.

i sadly do not know if that is possible im a datacenter technician not a dev, but my educated guess is that it should be possible... Remix uses vite under the hood if that helps 🤷

**Comment by duracell80** [comment]
The material design element of the UI is awesome, particularly for Android users. Try to keep that part of it for sure. Too many devs focus on the latest trends.

Some ideas:
- Make authentication easier, since it's self hosted LAN security could be a priority for some.

- Add URL's to the queue via a browser bookmarklet (if possible via an API with vanilla JavaScript?)

- Audio use cases: Download audio only in a separate section / Folder of the UI

**Reply by marcopiovanello** [reply]
Thanks for the suggestions! They're vital!

**Comment by ndom91** [comment]
Just wanted to mention there is a pretty great shadcn port to svelte, `shadcn-svelte` (https://shadcn-svelte.com)

**Comment by sdassow** [comment]
On a related note, I'd be interested in a native client using [Fyne](https://github.com/fyne-io/fyne).

**Comment by miaLDN** [comment]
Anything that isn’t based upon react.js

**Reply by marcopiovanello** [reply]
Why

**Reply by salzig** [reply]
@planetrocky it already uses react. Why bother switching?

**Reply by Aponnie** [reply]
> @planetrocky it already uses react. Why bother switching?

Oh... That _may_ explain some of the issues.

I for one am really not a desktop JS fan.  It's always seemed like the wrong tool for most jobs on the desktop, IMHO.

_Must_ this be JS-based?  It's just that all of the JS apps I've tried feel clunky to use and have weirdly functioning UI elements like it was made solely for a tablet.

Out of the many I came across over the recent years, none of them have stuck around for long, I eventually found a non-JS project that worked better - much more reliably and quickly. Sometimes, I just wrote my own solution.

(I am currently writing something similar to this project, but web-based for my file server at home.)

Could you write in a more conventional and reliable language?

**Comment by lamualfa** [comment]
Try mantine. It's popular, stable, consistent & have so much ready to use components.

https://mantine.dev/

**Comment by ffflick-gh** [comment]
Hello, is there any update to this? I am wondering when the next version with the updated UI will be released. Thanks 🙂

**Comment by TwoBrake** [comment]
You should check out [Chakra UI](https://chakra-ui.com/). A library that I recently started using, very useful hooks and components for React.

**Comment by Aponnie** [comment]
There's no option for it, unfortunately, but I think you should roll your own extensible UI specifically _for_ yt-dlp.

It will work far better than trying to make a general purpose framework fit this specific problem.

Also, issues will get fixed faster; changes to the Interface won't be potentially held back; many frameworks are notoriously slow; and you maintain ownership and control of the entire codebase.

It just seems less of a liability to me :-/

Just a thought.

"Define slow."
Slow to load, slow to respond, resource hungry, buggy, slow to fix.

Call me naïve, but development speed is not as important to me as the ability to use it, daily, without an issue.

A responsive web interface which is aware of the screen/viewport size would be a welcome minimum for most users: Spacious on desktop, compact and focused on mobile. (Plus, no weird menus that break usability by extending options off-screen!)

I could go on but I think I'll get my coat lol
